### ISSUE TITLE
suddenly, VSC doesnt recognize C# extensions

### ISSUE THREAD

**Issue by SporeProductions** [OmniSharp/omnisharp-vscode#3623]
So, I've been using VSCode for a few years now, primarily for C# and Unity. Was using it just fine today as I do every day, when it suddenly started complaining that it doesn't recognize ".csharp" and .cs files.

I have been using the official Microsoft c# extension.

The pop up to recommend an extension  after that error, has an install button. if I press that, the extensions panel shows that it searches for @id:ms-vscode.csharp and the results are "No Extensions Found"

My VScode was configured to auto-update VSC and extensions. I can only assume an update broke something.

I can still open .cs files and auto-complete of code seems to still work, but options like F12 (Go To Definition) cause a blue progress bar to animate at the top of VSC and nothing else happens.

I tried disabling extensions, restart VSC, rebooting the entire system, re-installing the c# extension... nothing helps.

I also deleted my Unity project's .vscode folder. Didn't help.

Omnisharp is running.

**Comment by daxtonb** [comment]
Same issue as [https://github.com/OmniSharp/omnisharp-vscode/issues/3622](here). Rolling back the version fixed the issue, so your assumption of an update is correct, as version 1.21.13 was published mid-day today.

**Comment by daxtonb** [comment]
Same issue as [#3622](https://github.com/OmniSharp/omnisharp-vscode/issues/3622#issue-576514851). Rolling back the version fixed the issue, so your assumption of an update is correct, as version 1.21.13 was published mid-day today.

**Comment by kabamjdenton** [comment]
This is happening to us as well and it reverts settings.

**Comment by BigGodMike** [comment]
> Same issue as [#3622](https://github.com/OmniSharp/omnisharp-vscode/issues/3622#issue-576514851). Rolling back the version fixed the issue, so your assumption of an update is correct, as version 1.21.13 was published mid-day today.

Yes, I try roll back to previous version, But VsCode just tell me that can not install ms-dotnettools.csharp and still the 1.21.13 is not fix the issue

**Comment by farchi** [comment]
@BigGodMike uninstall, download from https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.21.12 and install from VSIX (see screenshot).
Make sure to hit **Disable Auto Updating Extensions** if you see that option in the same menu so it stays at the correct version.
![image](https://user-images.githubusercontent.com/24467500/76043119-514b0d80-5f35-11ea-90d3-b154193a949a.png)

**Comment by BigGodMike** [comment]
@farchi Yes, Man, your screenshot help me directly to fix the issue ,Thanks a lot

**Comment by zhaolihang** [comment]
thanks

**Comment by chayZeng** [comment]
> @BigGodMike uninstall, download from https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.21.12 and install from VSIX (see screenshot).
> Make sure to hit **Disable Auto Updating Extensions** if you see that option in the same menu so it stays at the correct version.
> ![image](https://user-images.githubusercontent.com/24467500/76043119-514b0d80-5f35-11ea-90d3-b154193a949a.png)

it works, thanks

**Comment by SporeProductions** [comment]
Thank you for the reollback instructions. I've reverted to the previous version of the extension. While this allows me to work, and have Intellisense back in working order, the "problem" isn't truly solved until either the Marketplace ID is reverted or VSCode is updated to look for the new extension ID. It shouldn't be prompting to install an extension that it can not find.

**Comment by DenisWASD** [comment]
@farchi thanks, you saved my code night <3

**Comment by EnricoMonese** [comment]
This also broke all the extensions that depend on this!

**Comment by cartermp** [comment]
Thanks for reporting. The issue is due to the publisher rename, which has broken the unity extension. There is a pull request out on the unity extension, and once that is merged and updates go through, it should be resolved. Closing as a duplicate of #3620

**Comment by Yogioo** [comment]
> @BigGodMike uninstall, download from https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.21.12 and install from VSIX (see screenshot). Make sure to hit **Disable Auto Updating Extensions** if you see that option in the same menu so it stays at the correct version. ![image](https://user-images.githubusercontent.com/24467500/76043119-514b0d80-5f35-11ea-90d3-b154193a949a.png)

Thank you, It can works now
